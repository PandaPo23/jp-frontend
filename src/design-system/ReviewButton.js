import system from '../utils/System';
import IconButton from './IconButton';

const ReviewButton = system('ReviewButton', {
  extend: IconButton,
  size: 16,
  color: 'misc.muted',
  name: 'review-icon',
  labelMargin: 2,
});

export default ReviewButton;
