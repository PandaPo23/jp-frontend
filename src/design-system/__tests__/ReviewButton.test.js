import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import ReviewButton from '../ReviewButton';

describe('ReviewButton component', () => {
  it('renders correctly', () => {
    const localProps = {
      label: 'Green Comments',
      color: 'green',
    };
    const wrapper = mount(<ReviewButton {...localProps} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
