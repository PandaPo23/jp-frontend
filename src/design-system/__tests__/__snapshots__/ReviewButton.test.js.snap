// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewButton component renders correctly 1`] = `
.emotion-5 {
  color: green;
  background-color: transparent;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  border: none;
  cursor: pointer;
}

.emotion-0 {
  min-width: 16px;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-order: 1;
  -ms-flex-order: 1;
  order: 1;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 16px;
}

.emotion-4 {
  background-color: transparent;
  font-size: 20px;
  margin-left: 8px;
  -webkit-order: 2;
  -ms-flex-order: 2;
  order: 2;
  font-weight: bold;
  text-transform: uppercase;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
}

<ReviewButton
  color="green"
  label="Green Comments"
  labelMargin={2}
  name="review-icon"
  size={16}
>
  <IconButton
    className="emotion-6"
    color="green"
    label="Green Comments"
    labelMargin={2}
    name="review-icon"
    size={16}
  >
    <Flex
      alignItems="center"
      as="button"
      bg="transparent"
      border="none"
      className="emotion-6"
      color="green"
      disabled={false}
      display="flex"
      flexDirection="row"
      flexWrap="nowrap"
      hoverable={true}
      justifyContent={false}
      nobr={true}
    >
      <button
        className="emotion-5"
        disabled={false}
      >
        <Icon
          name="review-icon"
          order={1}
          size={16}
        >
          <Base
            className="emotion-0"
            name="review-icon"
            order={1}
            size={16}
          >
            <Flex
              centered={true}
              className="emotion-0"
              display="flex"
              order={1}
            >
              <div
                className="emotion-1"
              >
                <svg
                  className="emotion-0"
                  name="review-icon"
                  order={1}
                  size={16}
                />
              </div>
            </Flex>
          </Base>
        </Icon>
        <Text
          bg="transparent"
          bold={true}
          disabled={false}
          fontSize={3}
          mb={false}
          ml={2}
          mr={false}
          mt={false}
          nobr={true}
          order={2}
          truncate={true}
          uppercase={true}
        >
          <div
            className="emotion-4"
            disabled={false}
          >
            Green Comments
          </div>
        </Text>
      </button>
    </Flex>
  </IconButton>
</ReviewButton>
`;
